walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,91 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        int TIME_WINDOW_IN_MINUTES = 60 * 24;
+        Integer[] minutesNoOptAttendees = new Integer[TIME_WINDOW_IN_MINUTES];
+        Integer[] minutesOptAttendees = new Integer[TIME_WINDOW_IN_MINUTES];
+
+        Arrays.fill(minutesNoOptAttendees, 1);
+        Arrays.fill(minutesOptAttendees, 1);
+
+        ArrayList<TimeRange> availableTimeRanges = new ArrayList<>();
+        ArrayList<Event> eventsToSet = new ArrayList<>(events);
+
+        Collections.sort(eventsToSet, Event.ORDER_BY_START); // N * log N
```

Comment:
I don't think you need to sort for your solution

Reply:
That makes sense lol.

Will fix!